azure-client-runtime/src/main/java/com/microsoft/azure/v2/PollStrategy.java

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.v2;
+
+import com.microsoft.rest.v2.http.HttpRequest;
+import com.microsoft.rest.v2.http.HttpResponse;
+import rx.Single;
+
+import java.io.IOException;
+
+/**
+ * An abstract class for the different strategies that an OperationStatus can use when checking the
+ * status of a long running operation.
+ */
+abstract class PollStrategy {
+    private long delayInMilliseconds;
+
+    /**
+     * Get the number of milliseconds to delay before sending the next poll request.
+     * @return The number of milliseconds to delay.
+     */
+    final long delayInMilliseconds() {
+        return delayInMilliseconds;
+    }
+
+    /**
+     * Set the delay in milliseconds to 0.
+     */
+    final void clearDelayInMilliseconds() {
+        this.delayInMilliseconds = 0;
+    }
+
+    /**
+     * Update the delay in milliseconds from the provided HTTP poll response.
+     * @param httpPollResponse The HTTP poll response to update the delay in milliseconds from.
+     */
+    final void updateDelayInMillisecondsFrom(HttpResponse httpPollResponse) {
+        final String retryAfterSecondsString = httpPollResponse.headerValue("Retry-After");
+        if (retryAfterSecondsString != null && !retryAfterSecondsString.isEmpty()) {
+            try {
+                final long responseDelayInMilliseconds = Long.valueOf(retryAfterSecondsString) * 1000;
+                delayInMilliseconds = Math.max(responseDelayInMilliseconds, AzureProxy.defaultDelayInMilliseconds());
+            }
+            catch (NumberFormatException ignored) {
+            }
+        }
+    }
+
+    /**
+     * Create a new HTTP poll request.
+     * @return A new HTTP poll request.
+     */
+    abstract HttpRequest createPollRequest();
+
+    /**
+     * Update the status of this PollStrategy from the provided HTTP poll response.
+     * @param httpPollResponse The response of the most recent poll request.
+     */
+    abstract void updateFrom(HttpResponse httpPollResponse) throws IOException;
+
+    /**
+     * Update the status of this PollStrategy from the provided HTTP poll response.
+     * @param httpPollResponse The response of the most recent poll request.
+     * @return A Completable that can be used to chain off of this operation.
+     */
+    abstract Single<HttpResponse> updateFromAsync(HttpResponse httpPollResponse);
+
+    /**
+     * Get whether or not this PollStrategy's long running operation is done.
+     * @return Whether or not this PollStrategy's long running operation is done.
```

Comment:
in general when the method description and return value description are same we just remove the method description and keep the other one.

Reply:
Removed @return.